src/entities/accounts/accounts.go

```diff
@@ -0,0 +1,86 @@
+package accounts
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/rodolfoalvesg/api-banking/api/src/db"
+	"github.com/rodolfoalvesg/api-banking/api/src/models"
+	"github.com/rodolfoalvesg/api-banking/api/src/security"
+)
+
+func CreateAccount(body io.ReadCloser) ([]models.Account, error) {
+	bodyRequest, err := ioutil.ReadAll(body)
+	if err != nil {
+		return nil, err
+	}
+
+	var account models.Account
+	err = json.Unmarshal(bodyRequest, &account)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Não sei se eu seguiria com isso daqui... seu domínio não sabe que o dado chegou como json, eu deixaria essa parte onde estava mesmo, no controller! Aqui eu faria apenas o que é referente ao seu domínio: Account. Eu deixaria a parte de converter `json` -> `account` no próprio controller. O que acha?

Outro ponto é que você pode simplificar o seu código da seguinte forma (no controller):

```suggestion
	var account models.Account
        if err := json.NewDecoder(r.Body).Decode(&account); err != nil {
                return nil, err
        }
```

Reply:
Vou analizar p/ entender melhor e aplicar.